streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -836,10 +836,27 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
                                                      final OptimizableRepartitionNodeBuilder<K1, V1> optimizableRepartitionNodeBuilder) {
 
 
-        final String repartitionTopic = repartitionTopicNamePrefix + REPARTITION_TOPIC_SUFFIX;
-        final String sinkName = builder.newProcessorName(SINK_NAME);
-        final String nullKeyFilterProcessorName = builder.newProcessorName(FILTER_NAME);
-        final String sourceName = builder.newProcessorName(SOURCE_NAME);
+        final String repartitionTopicName = repartitionTopicNamePrefix.endsWith(REPARTITION_TOPIC_SUFFIX) ?
```

Comment:
If the user supplies the topic name, should we be appending the suffix at all?

Reply:
@cpettitt-confluent we always add a `-repartition` for all the repartition topics generated or supplied by the user.  When the users supply a name, it's used as the base name.  We've detailed this in our doc and Javadoc as well

Reply:
Sounds good. We can mark this resolved.